src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -308,7 +308,9 @@ public ZipFile(File file, Charset charset) throws IOException
     }
 
     /**
-     * Returns the zip file comment, or null if none.
+     * Returns the zip file comment. If a comment does not exist or an error is
+     * encountered decoding the comment using the charset specified
+     * when opening the Zip file, then {@code null} is returned.
```

Comment:
I've previously discussed options with Lance around this issue and I agree with the proposal to specify that it returns null when the decoding fails.

(In passing, the casing of "zip file" is very inconsistent in the javadoc, it's "ZIP file" in some places, "zip file" in others, the change here uses "Zip file").

Reply:
> (In passing, the casing of "zip file" is very inconsistent in the javadoc, it's "ZIP file" in some places, "zip file" in others, the change here uses "Zip file").

Thank you for pointing out the discrepancy above.

Let's decide on which way to go and I will update  under a separate PR.

I would suggest "ZIP file" based on the PKWare APPNOTE.TXT, but the man pages for info-zip command _zip_ command uses "zip file"

Apache commons also uses "ZIP file" or "Zip archive"

Reply:
> I would suggest "ZIP file" based on the PKWare APPNOTE.TXT,

I think using this case would be good.

Reply:
> I would suggest "ZIP file" based on the PKWare APPNOTE.TXT, but the man pages for info-zip command _zip_ command uses "zip file"

"ZIP" is not an acronym so I assume this is why they use "zip" in their docs.